ddtrace/contrib/starlette/middleware.py

```diff
@@ -0,0 +1,132 @@
+import sys
+
+from ddtrace import tracer as global_tracer
+from ddtrace.constants import ANALYTICS_SAMPLE_RATE_KEY
+from ddtrace.ext import SpanTypes, http
+from ddtrace.http import store_request_headers, store_response_headers
+from ddtrace.propagation.http import HTTPPropagator
+from ddtrace.settings import config
+
+from ...compat import reraise
+from ...internal.logger import get_logger
+from .utils import guarantee_single_callable
+
+log = get_logger(__name__)
+
+STARLETTE_VERSION = "starlette.version"
+STARLETTE_SPEC_VERSION = "starlette.spec_version"
+
+
+def bytes_to_str(str_or_bytes):
+    return str_or_bytes.decode() if isinstance(str_or_bytes, bytes) else str_or_bytes
+
+
+def _extract_tags_from_scope(scope):
+    tags = {}
+
+    http_method = scope.get("method")
+    if http_method:
+        tags[http.METHOD] = http_method
+
+    if config.starlette.trace_query_string:
+        query_string = scope.get("query_string")
+        if len(query_string) > 0:
+            tags[http.QUERY_STRING] = bytes_to_str(query_string)
+    else:
+        query_string = None
+
+    server = scope.get("server")
+    if server and len(server) == 2:
+        port = server[1]
+        server_host = server[0] + (":" + str(port) if port is not None and port != 80 else "")
+        full_path = scope.get("root_path", "") + scope.get("path", "")
+        http_url = scope.get("scheme", "http") + "://" + server_host + full_path
+        tags[http.URL] = http_url
+
+    http_version = scope.get("http_version")
+    if http_version:
+        tags[http.VERSION] = http_version
+
+    scope_starlette = scope.get("starlette")
+
+    if scope_starlette and "version" in scope_starlette:
+        tags[STARLETTE_VERSION] = scope_starlette["version"]
+
+    if scope_starlette and "spec_version" in scope_starlette:
+        tags[STARLETTE_SPEC_VERSION] = scope_starlette["spec_version"]
+
+    return tags
+
+
+def _extract_headers(scope):
+    headers = scope.get("headers")
+    if headers:
+        # headers: (Iterable[[byte string, byte string]])
+        return dict((bytes_to_str(k), bytes_to_str(v)) for (k, v) in headers)
+    return {}
+
+
+class TraceMiddleware:
+    """
+    Starlette application middleware that traces the requests.
+    Built based on ASGI middleware.
+
+    Args:
+        app: The Starlette application.
+        tracer: Custom tracer. Defaults to the global tracer.
+    """
+
+    def __init__(self, app, tracer=global_tracer, distributed_tracing=True):
+        self.app = guarantee_single_callable(app)
+        self.tracer = tracer
+        self._distributed_tracing = distributed_tracing
```

Comment:
I don't see this argument used. I think below you are doing the right thing by using the integration config as the way to enable/disable distributed tracing.

Reply:
Should I set distributed tracing in the config to the distributing tracing parameter or would it be better to just use `self._distributed_tracing`? I noticed some integrations use 
`if self._distributed_tracing:`
while others use
`config.asgi.distributed_tracing`.

At the moment I added a fix to use self._distributed_tracing instead of what distributed tracing is set to in the config